refactor: 세션 상세 API 일시 필드 ISO 8601 표준으로 변경

eeun1225 opened the pull request:
## 📌 Related Issue


## 🚨 해결하려는 문제가 무엇인가요?
- 세션 시작/종료 날짜와 시간을 별도로 조합하지 않아도 되도록 API 응답을 개선합니다.

## ⭐️ 어떻게 해결했나요?
- `SessionDetailsResponseV2` DTO를 추가했습니다.
- `startDateTime`, `endDateTime` 필드를 `ZonedDateTime`으로 추가했습니다.
- `/v2/sessions/{sessionId}` 엔드포인트 추가했습니다.

## 🤔 어떤 부분에 집중하여 리뷰해야 할까요?
- `findSessionDetails()`와 `findSessionDetailsV2()`의 공통 로직을 별도 함수로 분리하는 것이 좋을까요?

## 🗒️ 참고자료


## RCA 룰
- R: 꼭 반영해 주세요. 적극적으로 고려해 주세요. (Request changes)
- C: 웬만하면 반영해 주세요. (Comment)
- A: 반영해도 좋고 넘어가도 좋습니다. 그냥 사소한 의견입니다. (Approve)

eeun1225 committed:
refactor: ISO 8601 표준 세션 상세 조회 API 추가 (#174)

- SessionDetailsResponseV2 DTO 추가
- startDateTime, endDateTime 필드로 날짜/시간 통합
- /v2/sessions/{sessionId} 엔드포인트 추가

eeun1225 committed:
refactor: SessionDetailsResponseV2 타임존 정보 추가 (#174)

- LocalDateTime에서 ZonedDateTime으로 변경
- +09:00 타임존 정보 포함

devmizz commented on src/main/kotlin/co/yappuworld/schedule/client/dto/response/SessionDetailsResponseV2.kt:
```diff
@@ -0,0 +1,67 @@
+    @field:Schema(description = "세션 제목", example = "개발 세션")
+    val title: String,
+    @field:Schema(description = "세션 시작 일시", example = "2024-01-01T09:00:00+09:00")
+    val startDateTime: ZonedDateTime,
```
C: LocalDateTime이 아닌 ZonedDateTime으로 한 이유가 있을까요? 시간대 정보가 필요할 거 같진 않고, 어드민 쪽 응답 일부에서 이미 LocalDateTime 타입을 사용하고 있어요!

eeun1225 replied:
앱 쪽에서 2001-01-01T00:00:00+09:00 형태의 응답을 원하시는 것 같아서 ZonedDateTime을 사용했습니다!
타임존 정보가 필요 없다면 LocalDateTime으로 변경하겠습니다~!

devmizz replied:
넵! 타임존은 필요 없을 거 같아요! LocalDateTime으로 변경해주세요!

devmizz commented on src/main/kotlin/co/yappuworld/schedule/client/presentation/ScheduleApi.kt:
```diff
@@ -402,4 +403,12 @@ interface ScheduleApi {
     ): ResponseEntity<SuccessResponse<SessionDetailsResponse>>
+
+    @Operation(summary = "세션 상세 조회 (ISO 8601)")
+    @ApiResponses()
```
R: 혹시 여기 비워두신 이유가 있을까요?

eeun1225 replied:
정상 응답은 이미 DTO에서 @Schema를 통해 문서화되고 있어서 따로 정의할 필요가 없다고 생각했습니다!

devmizz replied:
아 저도 비워놨군요? 오호... 성공 케이스와 실패 케이스, 성공 케이스에서도 발생할 수 있는 여러가지 케이스들을 명시할 수 있다는 점에서 채워둬야 하긴 하는데...
스웨거 작성 연습도 하실 겸, ScheduleError.NO_WRITER_FOR_SCHEDULE 에러 관련해서만 한 번 추가해보실래요?

devmizz commented on src/main/kotlin/co/yappuworld/schedule/client/presentation/ScheduleApi.kt:
```diff
@@ -402,4 +403,12 @@ interface ScheduleApi {
         @PathVariable sessionId: UUID
     ): ResponseEntity<SuccessResponse<SessionDetailsResponse>>
+
+    @Operation(summary = "세션 상세 조회 (ISO 8601)")
```
C: summary에는 API 자체에 대한 설명이 들어가야 한다고 생각하는데, ISO 8601은 v1과의 차별점이라서 적절치 않다고 생각해요! 만약 다른 수정 사항이 생기면 summary에 또 다른 설명을 추가할 수는 없으니까요!

eeun1225 replied:
아.. 그렇군요 😭
그러면 summary에는 v1과 동일하게 세션 상세 조회로 두고 description으로 차별점을 작성하는게 맞을까요?

devmizz replied:
전혀 중요한 부분은 아니긴 한데, 그게 조금 더 명확하고 좋을 거 같아요! 제 주관이라 C는 아니고 A로 하는게 맞았겠네요. 고려해보시고 정은님이 생각하시기에 적절한 방향으로 수정해주세요!

devmizz reviewed (changes requested):
개인적으로는 기존 응답에 startDateTime, endDateTime 필드만 추가하는 걸 생각하긴 했었는데, 이 방법도 크게 문제가 될 거 같진 않습니다.

devmizz commented:
> findSessionDetails()와 findSessionDetailsV2()의 공통 로직을 별도 함수로 분리하는 것이 좋을까요?

이 부분을 이제야 봤네요.
개인적으로는 로직이 크게 달라질 부분이 없어서 합쳐도 될 거 같은데, 나중에 최소 버전을 올리면서 v1을 deprecated 하면 불필요한 private method가 생기는 결과로 이어질 거 같아요!
결과적으로는 크게 상관 없을 거 같습니다! 나중에 deprecated 할 때 private method는 지워도 되는 거니깐요. 나름의 판단 근거만 갖고 작성하시면 될 거 같아요!

eeun1225 committed:
refactor: 세션 상세 조회 V2 API 수정 (#174)

- ZonedDateTime을 LocalDateTime으로 변경
- @Operation description 추가 (ISO 8601 형식 명시)
- NO_WRITER_FOR_SCHEDULE 에러 응답 문서화

devmizz commented on src/main/kotlin/co/yappuworld/schedule/client/dto/response/SessionDetailsResponseV2.kt:
```diff
@@ -0,0 +1,65 @@
+    val progressPhase: SessionProgressPhase,
+    @field:Schema(description = "세션 제목", example = "개발 세션")
+    val title: String,
+    @field:Schema(description = "세션 시작 일시", example = "2024-01-01T09:00:00+09:00")
```
R: example 수정이 필요해보여요!

eeun1225 committed:
fix: SessionDetailsResponseV2 example 타임존 정보 제거 (#174)
